disnake/ui/action_row.py

```diff
@@ -24,86 +24,162 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, List, Optional, Sequence, TypeVar, Union
-
-from ..components import ActionRow as ActionRowComponent, NestedComponent, SelectOption
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ClassVar,
+    Generic,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    TypeVar,
+    Union,
+    overload,
+)
+
+from ..components import (
+    ActionRow as ActionRowComponent,
+    Button as ButtonComponent,
+    NestedComponent,
+    SelectMenu as SelectComponent,
+    SelectOption,
+)
 from ..enums import ButtonStyle, ComponentType, TextInputStyle
 from ..utils import MISSING
 from .button import Button
 from .item import WrappedComponent
 from .select import Select
 from .text_input import TextInput
 
-__all__ = ("ActionRow",)
-
 if TYPE_CHECKING:
     from ..emoji import Emoji
+    from ..message import Message
     from ..partial_emoji import PartialEmoji
     from ..types.components import ActionRow as ActionRowPayload
 
-    ActionRowT = TypeVar("ActionRowT", bound="ActionRow")
-    Components = Union[
-        ActionRowT,
-        WrappedComponent,
-        Sequence[Union[ActionRowT, WrappedComponent, Sequence[WrappedComponent]]],
-    ]
+__all__ = ("ActionRow",)
+
 
+MessageUIComponent = Union[Button[Any], Select[Any]]
+ModalUIComponent = Union[TextInput, Select[Any]]
+UIComponentT = TypeVar("UIComponentT", bound=WrappedComponent)
+StrictUIComponentT = TypeVar("StrictUIComponentT", MessageUIComponent, ModalUIComponent)
 
-class ActionRow:
+Components = Union[
+    "ActionRow[UIComponentT]",
+    UIComponentT,
+    Sequence[Union["ActionRow[UIComponentT]", UIComponentT, Sequence[UIComponentT]]],
+]
```

Comment:
Thoughts on exposing these somewhere? Would make typehints easier than the current style: 
```py
List[disnake.ui.ActionRow[disnake.ui.action_row.MessageUIComponent]]
```

Reply:
Good idea, done ^^